repair/repair.cc

```diff
@@ -562,6 +562,7 @@ repair_info::repair_info(repair_service& repair,
     , gossiper(repair.get_gossiper())
     , sharder(get_sharder_for_tables(db, keyspace_, table_ids_))
     , keyspace(keyspace_)
+    , erm(db.local().find_keyspace(keyspace).get_effective_replication_map())
```

Comment:
This is good, but forces us to consider several paths:

1. we keep erm and block topology changes until all its users are gone - so repair blocks topology changes
2. we add erm->transitioning_away_from() and sample it regularly during repair, failing or restarting repair if it happens. This allows topology changes to preempt repair
3. some mix where we complete a partial repair and then restart it using the new topology

Anyway, this is okay for now, but we rely on the operator not to change topology while repair is running.

Reply:
Cc @asias @tgrabiec